### ISSUE TITLE
bug: datetime: Unable to pick time when using ion-datetime in modal on firefox

### ISSUE THREAD

**Issue by PMudra** [ionic-team/ionic-framework#24585]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

When placing the ion-datetime inside a modal, we have noticed a bug when using firefox.
The time picker popover of the datetime picker sometimes is moved behind the datetime modal which makes it hard to pick a time.
This behavior does not occur when using ion-datetime without a modal.
Other browsers seem to work (tested with Chrome for example).

<details><summary>Gif demo</summary>

![Peek 2022-01-17 12-38](https://user-images.githubusercontent.com/84765/149763217-c35f7963-883a-40e9-a871-3304b902df3d.gif)

</details>

### Expected Behavior

The time picker should stay in front when trying to pick a time.
Use Chrome to see expected behavior.

### Steps to Reproduce

1. Open ion-datetime example with firefox: https://ionicframework.com/docs/api/datetime 
2. Scroll down to and click the "Open Datetime Modal" button to open the modal
3. Click on the time to pick a new time
4. Wait, move the mouse (mostly when moving down?) or click on the time again

The popover will suddenly be moved to the background / disappear.

### Code Reproduction URL

https://ionicframework.com/docs/api/datetime

### Ionic Info

Testing via official example page. Current latest version is v6.0.2.

### Additional Information

Also note that the height of the datetime content behaves differently on firefox than on Chrome (md version only for some reason). The time info and button are cut off in the firefox demo.

**Comment by sean-perkins** [comment]
@PMudra thanks for reporting this issue! For the datetime examples, we do set an explicit height on the modal and the datetime within it. This may just be a display issue that needs to be fixed in the example. 

It's important to note that this only happens in the Android preview (`md` mode) of the demo.

**Comment by sean-perkins** [comment]
Upon a further triage, it looks like the `.calendar-day` (individual days of the month) is being sized differently between Chrome and Firefox.

On chrome, the button will be 38.56px x 42.5px, but on Firefox it is 38.56px x 43.5px. The extra pixel affordance on each row is causing the time selector to be pushed off the view. 

This may be a regression from #24522, adding top padding to grid container.

**Comment by JessiPio4cronn** [comment]
@sean-perkins thanks for the fast reply. I'm from the same team as @PMudra.
We already adjusted the height in our own setup, so that you can see all the buttons, but the behavior with the popover stays the same. I attached a Gif demo of our setup, where the behavior is better visible. In the demo, the issue occurs on the second try.

![firefox-datetime-time-popover-issue](https://user-images.githubusercontent.com/62110002/150173133-e3671dc2-9777-4eb2-9acd-798ac2c6fefc.gif)

**Comment by laurentc123** [comment]
Hi, 
I have the same problem.
reproduced on https://github.com/laurentc123/ionic6-datetime-modal

click button to open the modal, click to open the calendar.
Click on time to modify time -> the ion-datetime disapears behind the modal.
I found the problem only on firefox, ok on Chrome
Thank you !

**Comment by bastiaan-stuck** [comment]
Greetings,
I am facing the same issue, both in the demo in the docs on Android/md mode as well as in our application on Firefox.
Chrom[e|ium] works ok.

Not sure if this is related to `.calendar-day`, that only seems push the row with the button that opens the modal out of view. My issue, as demonstrated in the 2 GIFs posted by others in this issue, is the time selector popover appearing behind the modal the `ion-date-time` is inside of, making time selection itself very clunky.

I have attached another GIF, please note the erraticness when hovering/clicking on the time picker:
![date-time-popover-bug](https://user-images.githubusercontent.com/88327349/150380203-c73d259d-b6a9-49c1-b1d3-5aef83c81145.gif)

**Comment by sean-perkins** [comment]
Can anyone experiencing this issue test with the this dev build and let me know if the problem persists?

```bash
npm install @ionic/core@6.0.4-dev.1642712337.fed9e8f
```

Thanks!

**Comment by laurentc123** [comment]
Hi @sean-perkins , thank you for your reply.
I'm still experiencing the problem with the version 6.0.4-dev.1642712337.fed9e8f
I've updated my app https://github.com/laurentc123/ionic6-datetime-modal with  this version

Thanks !

**Comment by sean-perkins** [comment]
@laurentc123 thank you so much for testing and attaching an image for reference 👍 

I've updated the build and confirmed the issue does not appear anymore in your example app:
```
npm install @ionic/angular@6.0.4-dev.1642791040.fed9e8f --save
```

Dev builds should install their associated peer dependencies for `@ionic/core` with the correct version (without needing to manually specify).

Firefox (with design mode)
<video src="https://user-images.githubusercontent.com/13732623/150584770-aff8727a-4ec3-4b6d-9d4e-7c4fbb86f675.mp4"></video>

Firefox (without design mode)
<video src="https://user-images.githubusercontent.com/13732623/150585003-a77c84ce-ebf5-4058-86fd-a5033f5e87e4.mp4"></video>

I'll look into the flicker issue separately (seen in the videos). For some reason on first opening the modal and toggling the popover overlay quickly, the modal will move on top of the popover and then repaint correctly under.

Let me know if you see the same 👍

**Comment by liamdebeasi** [comment]
This issue was resolved in Ionic 6.0.4 but the ticket was never closed out. I am going to close this, but please open a new issue if you run into any other problems. Thanks!
